### ISSUE TITLE
No terms could be added bug

### ISSUE THREAD

**Issue by jacobrh91** [ncsa/NCSA-Genomics_EpiQuant_SEMS_Spark#20]
When running:

```
spark-submit --master local[2] target/SPAEML-0.0.1-jar-with-dependencies.jar StepwiseModelSelection --epiq src/test/resources/Genotypes/10.Subset.n\=2648.epiq -P src/test/resources/Phenotypes/Simulated.Data.10.Reps.Herit.0.92_n\=2648.epiq -o output.txt
```

Got the following error:

```
18/11/05 14:01:17 INFO SparkContext: Running Spark version 2.1.0
18/11/05 14:01:18 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
18/11/05 14:01:18 INFO SecurityManager: Changing view acls to: jacobheldenbrand
18/11/05 14:01:18 INFO SecurityManager: Changing modify acls to: jacobheldenbrand
18/11/05 14:01:18 INFO SecurityManager: Changing view acls groups to: 
18/11/05 14:01:18 INFO SecurityManager: Changing modify acls groups to: 
18/11/05 14:01:18 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(jacobheldenbrand); groups with view permissions: Set(); users  with modify permissions: Set(jacobheldenbrand); groups with modify permissions: Set()
18/11/05 14:01:18 INFO Utils: Successfully started service 'sparkDriver' on port 63493.
18/11/05 14:01:18 INFO SparkEnv: Registering MapOutputTracker
18/11/05 14:01:18 INFO SparkEnv: Registering BlockManagerMaster
18/11/05 14:01:18 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
18/11/05 14:01:18 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
18/11/05 14:01:18 INFO DiskBlockManager: Created local directory at /private/var/folders/n0/wzblr5sd0k93z7_xbkxnbwc80000gn/T/blockmgr-660bfdd3-d8b0-45c4-bf45-07abdaeed909
18/11/05 14:01:18 INFO MemoryStore: MemoryStore started with capacity 366.3 MB
18/11/05 14:01:18 INFO SparkEnv: Registering OutputCommitCoordinator
18/11/05 14:01:19 INFO Utils: Successfully started service 'SparkUI' on port 4040.
18/11/05 14:01:19 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://10.193.232.204:4040
18/11/05 14:01:19 INFO SparkContext: Added JAR file:/Users/jacobheldenbrand/git/Genomics_EpiQuant/SPAEML/target/SPAEML-0.0.1-jar-with-dependencies.jar at spark://10.193.232.204:63493/jars/SPAEML-0.0.1-jar-with-dependencies.jar with timestamp 1541448079133
18/11/05 14:01:19 INFO Executor: Starting executor ID driver on host localhost
18/11/05 14:01:19 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 63494.
18/11/05 14:01:19 INFO NettyBlockTransferService: Server created on 10.193.232.204:63494
18/11/05 14:01:19 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
18/11/05 14:01:19 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 10.193.232.204, 63494, None)
18/11/05 14:01:19 INFO BlockManagerMasterEndpoint: Registering block manager 10.193.232.204:63494 with 366.3 MB RAM, BlockManagerId(driver, 10.193.232.204, 63494, None)
18/11/05 14:01:19 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 10.193.232.204, 63494, None)
18/11/05 14:01:19 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 10.193.232.204, 63494, None)
18/11/05 14:01:19 INFO SharedState: Warehouse path is 'file:/Users/jacobheldenbrand/git/Genomics_EpiQuant/SPAEML/spark-warehouse/'.
Iteration number: 1
The Response variable: Heritability_0.92_Rep_1

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037131   -0.049065     0.010275    -4.775   0.000002
(Intercept)                 1.095971     0.008000   137.001   0.000000

Iteration number: 2
The Response variable: Heritability_0.92_Rep_1

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037131   -0.032158     0.012517    -2.569   0.010246
S1_10036985_S1_10037131    -0.032933     0.013949    -2.361   0.018300
(Intercept)                 1.110072     0.009978   111.253   0.000000

Iteration number: 3
The Response variable: Heritability_0.92_Rep_1

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037131    0.008003     0.021089     0.380   0.704347
S1_10036985_S1_10037131    -0.057355     0.017346    -3.307   0.000957
S1_100302959_S1_10037732   -0.047150     0.019937    -2.365   0.018102
(Intercept)                 1.139846     0.016059    70.981   0.000000

Iteration number: 4
The Response variable: Heritability_0.92_Rep_1

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037732   -0.045976     0.012146    -3.785   0.000157
S1_10036985_S1_10037131    -0.051497     0.011440    -4.501   0.000007
S1_100302919_S1_10037702   -0.069679     0.039854    -1.748   0.080519
(Intercept)                 1.140653     0.013423    84.978   0.000000

Calculation time (in seconds): 7.177859204
Calculation time (in minutes): 0.11963098673333333
Calculation time (in hours): 0.001993849778888889

Genotype File: src/test/resources/Genotypes/10.Subset.n=2648.epiq
Phenotype File: src/test/resources/Phenotypes/Simulated.Data.10.Reps.Herit.0.92_n=2648.epiq

The Response variable: Heritability_0.92_Rep_1

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037732   -0.041058     0.011820    -3.474   0.000522
S1_10036985_S1_10037131    -0.052405     0.011433    -4.584   0.000005
(Intercept)                 1.136379     0.013203    86.067   0.000000


-	df	SS	MS	F
S1_100302959_S1_10037732	1	0.8017	0.8017	12.06564	0.00052	
S1_10036985_S1_10037131	1	1.3961	1.3961	21.01068	0.00000	
Error	2645	175.7533	0.0664	-	-
Total	2647	178.0077	89.0039	16.96359	0.00000	
Iteration number: 1
The Response variable: Heritability_0.92_Rep_10

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037732   -0.055910     0.012257    -4.561   0.000005
S1_10036985_S1_10037131    -0.050903     0.011508    -4.423   0.000010
S1_100246792                0.087482     0.030714     2.848   0.004429
(Intercept)                 1.061888     0.030958    34.300   0.000000

Iteration number: 2
The Response variable: Heritability_0.92_Rep_10

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100246792                0.087494     0.030705     2.849   0.004413
S1_100302959_S1_10037732   -0.042290     0.015032    -2.813   0.004940
S1_10036985_S1_10037131    -0.045519     0.012009    -3.791   0.000154
S1_100302959_S1_10037702   -0.020180     0.012899    -1.564   0.117830
(Intercept)                 1.058000     0.031050    34.075   0.000000

Calculation time (in seconds): 3.719731121
Calculation time (in minutes): 0.061995518683333334
Calculation time (in hours): 0.0010332586447222222

Genotype File: src/test/resources/Genotypes/10.Subset.n=2648.epiq
Phenotype File: src/test/resources/Phenotypes/Simulated.Data.10.Reps.Herit.0.92_n=2648.epiq

The Response variable: Heritability_0.92_Rep_10

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100302959_S1_10037732   -0.055910     0.012257    -4.561   0.000005
S1_10036985_S1_10037131    -0.050903     0.011508    -4.423   0.000010
S1_100246792                0.087482     0.030714     2.848   0.004429
(Intercept)                 1.061888     0.030958    34.300   0.000000


-	df	SS	MS	F
S1_100302959_S1_10037732	1	1.3963	1.3963	20.80561	0.00001	
S1_10036985_S1_10037131	1	1.3131	1.3131	19.56506	0.00001	
S1_100246792	1	0.5445	0.5445	8.11274	0.00443	
Error	2644	177.4481	0.0671	-	-
Total	2647	180.5367	60.1789	15.33989	0.00000	
Iteration number: 1
The Response variable: Heritability_0.92_Rep_100

Name                        Estimate   Std. Error   t value   Pr(>|t|)
S1_100246792                0.073967     0.030662     2.412   0.015918
S1_100302959_S1_10037732   -0.153787     0.072417    -2.124   0.033793
S1_10036985_S1_10037131    -0.048692     0.011550    -4.216   0.000026
S1_100302959                0.108593     0.072885     1.490   0.136363
(Intercept)                 1.063701     0.030985    34.330   0.000000

Exception in thread "main" java.lang.Exception: No terms could be added to the model at a cutoff of 0.05
	at spaeml.SPAEML$.performSteps(SPAEML.scala:290)
	at spaeml.SPAEML$$anonfun$performSPAEML$1.apply(SPAEML.scala:506)
	at spaeml.SPAEML$$anonfun$performSPAEML$1.apply(SPAEML.scala:502)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at spaeml.SPAEML$.performSPAEML(SPAEML.scala:502)
	at StepwiseModelSelection$.launch(StepwiseModelSelection.scala:131)
	at Main$.main(Main.scala:16)
	at Main.main(Main.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:738)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:187)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:212)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:126)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

This error is only supposed to happen when there are no termed to be added on the first iteration. Clearly, this was not the case here.

This was found after Jacob deleted the Sparse implementation, so its possible that this introduced a bug

**Comment by jacobrh91** [comment]
Solved the problem. It turns out that we are processing multiple phenotypes, and we get an error when looking at the third phenotype claiming that no more terms could be added.

But it turns out that when looking at multiple phenotypes, the terms included in the model are not reset between phenotypes! I am preparing a fix for this.

This probably results from testing typically occurring with single phenotypes only

**Comment by jacobrh91** [comment]
Fix part of PR#21
